templates/config.toml

```diff
@@ -38,56 +38,53 @@ oom_score = 0
       no_pivot = false
       [plugins.cri.containerd.default_runtime]
         runtime_type = "io.containerd.runtime.v1.linux"
-	  {% if untrusted %}
+      {%- if untrusted %}
       [plugins.cri.containerd.untrusted_workload_runtime]
         runtime_type= "io.containerd.{{ untrusted_name }}.v2"
-	  {% endif %}
+      {% endif %}
       [plugins.cri.containerd.runtimes]
         [plugins.cri.containerd.runtimes.runc]
           runtime_type = "io.containerd.runc.v1"
-	    {% if untrusted %}
+        {%- if untrusted %}
         [plugins.cri.containerd.runtimes.{{ untrusted_name }}]
           runtime_type= "io.containerd.{{ untrusted_name }}.v2"
           [plugins.cri.containerd.runtimes.{{ untrusted_name }}.options]
             Runtime = "{{ untrusted_binary }}"
             RuntimeRoot = "{{ untrusted_path }}"
-		{% endif %}
+        {% endif %}
     [plugins.cri.cni]
       bin_dir = "/opt/cni/bin"
       conf_dir = "/etc/cni/net.d"
       conf_template = ""
     [plugins.cri.registry]
       [plugins.cri.registry.mirrors]
-    {% if custom_registries -%}
-      {% for registry in custom_registries -%}
-        {% if registry.host -%}
+      {%- if custom_registries %}
+      {%- for registry in custom_registries %}
+        {%- if registry.host %}
         [plugins.cri.registry.mirrors."{{ registry.host }}"]
-          {% if registry.url -%}
+          {%- if registry.url %}
           endpoint = ["{{ registry.url}}"]
-          {% endif -%}
+          {%- endif -%}
         {% endif -%}
-      {% endfor -%}
-    {% endif -%}
-    {% if custom_registries %}
-      [plugins.cri.registry.auths]
-      {% for registry in custom_registries %}
-        {% if registry.username and registry.password  %}
-        [plugins.cri.registry.auths."{{ registry.url }}"]
+      {% endfor%}
+      [plugins.cri.registry.configs]
+      {%- for registry in custom_registries %}
+        {%- if registry.username and registry.password %}
+        [plugins.cri.registry.configs."{{ registry.url }}".auth]
```

Comment:
I don't think the change from `plugins.cri.registry.auths."{{ registry.url }}"` to `plugins.cri.registry.configs."{{ registry.url }}".auth` is correct for v1 config. See:

https://github.com/containerd/containerd/blob/7ddf5e52ba738e868b70807797c79c8e54da3497/docs/cri/registry.md

Which directs you here for v1 config:

https://github.com/containerd/cri/blob/release/1.2/docs/registry.md

Which uses the `plugins.cri.registry.auths."{{ registry.url }}"` format.

I'm going to need more details before I can approve this. Why is this change correct?

Reply:
do you agree it's correct for the v2 config [here](https://github.com/charmed-kubernetes/charm-containerd/pull/59/files/597d62d6f457776276cd1abded46ed53e55224d0#diff-2d091891ba67d052f96d138d6604740ada259fc6c54dc44f5cb9f53ff7d1dac9R72-R74)?  

Much of this PR was a derivation of the work on #52.  If I recall, I spent quite some time looking at the go parser for the configs and adjusted the templates and tests to work together.  Granted, the integration tests do only (currently) cover the `1.22/stable` release, so perhaps these v1 configs do not function the same in older releases.  Is that the issue here?

Please see my reasoning for these changes [here](https://github.com/charmed-kubernetes/charm-containerd/pull/52/files#r740460812)

Reply:
Agreed that the change for v2 config is correct, for sure.

Thanks for the background. Maybe the change for v1 is fine too. Let me come back to this when I can think a little more clearly.